ocaml/xapi/xapi_vtpm.ml

```diff
@@ -20,19 +20,41 @@ let assert_no_vtpm_associated ~__context vm =
       let amount = List.length vtpms |> Int.to_string in
       raise Api_errors.(Server_error (vtpm_max_amount_reached, [amount]))
 
-let create ~__context ~vM ~is_unique =
-  assert_no_vtpm_associated ~__context vM ;
-  Xapi_vm_lifecycle.assert_initial_power_state_is ~__context ~self:vM
-    ~expected:`Halted ;
+let introduce ~__context ~vM ~persistence_backend ~contents ~is_unique =
   let ref = Ref.make () in
   let uuid = Uuid.(to_string (make ())) in
   let backend = Ref.null in
-  let persistence_backend = `xapi in
-  let contents = Xapi_secret.create ~__context ~value:"" ~other_config:[] in
   Db.VTPM.create ~__context ~ref ~uuid ~vM ~backend ~persistence_backend
     ~is_unique ~is_protected:false ~contents ;
   ref
 
+(** Contents from unique vtpms cannot be copied! *)
+let get_contents ~__context ?from () =
+  let block_copy ref =
+    if Db.VTPM.get_is_unique ~__context ~self:ref then None else Some ref
```

Comment:
I am confused. We are either returning `None` or the argument as `Some`. This suggests the function is a predicate on `ref` but this not reflected in the name.

Reply:
I was afraid it might cause confusion.
The purpose of this function is to disallowing VTPMs with the unique profile from ever being copied while being able to compose with the optional parameter `?frpm` which contains a `ref option`.

I'll try thinking of a less confusing way to express this

Reply:
Is this version easier to follow?